type.go

```diff
@@ -399,6 +402,48 @@ func (t int32Type) AssignValue(dst reflect.Value, src Value) error {
 		dst.SetInt(int64(v))
 	case reflect.Uint8, reflect.Uint16, reflect.Uint32:
 		dst.SetUint(uint64(v))
+	case reflect.Pointer:
+		underlying := reflect.TypeOf(dst).Elem().Kind()
+		switch underlying {
+		case reflect.Float32:
+			val := float32(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Float64:
+			val := float64(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Int8:
+			val := int8(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Int16:
+			val := int16(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Int32:
+			dst.Set(reflect.ValueOf(&v))
+		case reflect.Int64:
+			val := int64(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Uint8:
+			val := uint8(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Uint16:
+			val := uint16(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Uint32:
+			val := uint32(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Uint64:
+			val := uint64(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Int:
+			val := int(v)
+			dst.Set(reflect.ValueOf(&val))
+		case reflect.Uint:
+			val := uint(v)
+			dst.Set(reflect.ValueOf(&val))
+		default:
+			// trigger a panic for type correction
+			dst.Set(reflect.ValueOf(v))
+		}
```

Comment:
Would it be possible to simplify this code so we don't repeat the conversion rules twice?

Reply:
refactored the code.